pipelines/deployment-pipelines/wait-for-workloads-readiness-or-fail/pipeline.yaml

```diff
@@ -0,0 +1,47 @@
+name: Wait for Workloads readiness or fail
+
+# This pipeline is a deployment request pipeline and runs for any deploy or re-deploy.
+on:
+  deployment_request: {}
+
+# This pipeline needs permissions to deploy to our development environments. We could add staging and production here too.
+permissions:
+  application: developer
+  env-types:
+    development: deployer
+
+# Because this pipeline runs tests, we're going to use concurrency controls to ensure only one deploy and test cycle is running at a time against the target environment.
+concurrency:
+  group: "${{ pipeline.id }}-${{ pipeline.app.id }}-${{ inputs.env_id }}"
+
+jobs:
+  deploy:
+    steps:
+    - if: ${{ ! inputs.set_id }}
+      name: Create Deployment Set
+      id: create-deployment-set
+      uses: actions/humanitec/apply@v1
+      with:
+        delta_id: ${{ inputs.delta_id }}
+        env_id: ${{ inputs.env_id }}
+
+    - name: Deploy Set To Environment
+      uses: actions/humanitec/deploy@v1
+      with:
+        set_id: ${{ inputs.set_id || steps.create-deployment-set.outputs.set_id }}
+        value_set_version_id: ${{ inputs.value_set_version_id }}
+        env_id: ${{ inputs.env_id }}
+        message: ${{ inputs.comment }}
+
+    - name: Wait for Workloads readiness
+      continue-on-error: true
+      id: wait-for-workloads-readiness
+      uses: actions/humanitec/wait-for-readiness@v1
+      with:
+        env_id: ${{ inputs.env_id }}
+
+    - if: ${{ steps.wait-for-workloads-readiness.status == 'failed' }}
+      name: Fail if Workloads readiness failed

```

Comment:
Can you add a comment explaining why this is broken out to a separate step from the previous one?

Reply:
I added something, but just to make sure, why do we need this again? Failure of the previous step (with timeout) won't be enough by itself?

Reply:
@mathieu-benoit It should be fine by itself, the extra step is just to add a custom message or some slack notification or something.

Reply:
Gotcha, I'll update the comment on top of this step to clarify this. Thanks.

Reply:
Done.